lib/fog/ovirt/requests/compute/v4/create_vm.rb

```diff
@@ -9,18 +9,40 @@ def check_for_option(opts, name)
 
           # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
           def process_vm_opts(opts)
-            return unless check_for_option(opts, "template") && check_for_option(opts, "storagedomain")
+            return unless check_for_option(opts, "template") && check_for_option(opts, :disks)
 
-            template_id = opts[:template] || client.system_service.templates_service.search(:name => opts[:template_name]).first.id
-            template_disks = client.system_service.templates_service.template_service(template_id).get.disk_attachments
-            storagedomain_id = opts[:storagedomain] || storagedomains.select { |s| s.name == opts[:storagedomain_name] }.first.id
+            template_id = opts[:template].id || client.system_service.templates_service.search(:name => opts[:template_name]).first.id
+            template_disks = client.system_service.templates_service.template_service(template_id).disk_attachments_service.list
 
             # Make sure the 'clone' option is set if any of the disks defined by
             # the template is stored on a different storage domain than requested
             opts[:clone] = true unless opts[:clone] == true || template_disks.empty? || template_disks.all? { |d| d.storage_domain == storagedomain_id }
 
-            # Create disks map
-            opts[:disks] = template_disks.collect { |d| { :id => d.id, :storagedomain => storagedomain_id } }
+
+            # Create disks map "clone" from disks, rather then template
+            if opts[:disks] && !opts[:disks].empty?
+              disk_attachments = []
+              opts[:disks].each do |disk|
+                sparse = disk["sparse"].present? ? disk["sparse"] == "true" : true
```

Comment:
Since the PR https://github.com/fog/fog-ovirt/pull/41 was merged, the Boolean values should stay Boolean and not string.